src/WebJobs.Script.WebHost/Middleware/SystemTraceMiddleware.cs

```diff
@@ -36,6 +37,11 @@ public async Task Invoke(HttpContext context)
 
             await _next.Invoke(context);
 
+            if (context.RequestAborted.IsCancellationRequested)
```

Comment:
```suggestion
            if (context.RequestAborted.IsCancellationRequested && !context.Response.HasStarted)
```

Got this from the PR which added this handling to AspNetCore: https://github.com/dotnet/aspnetcore/pull/46330

May be useful to skim this PR to see if there is anything else we should be handling/doing

Reply:
I wonder if the changes here might be greater and it's worth its own PR?